### PR TITLE
EVG-12752: Add data fetching for system metrics in rest package 

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: rest/data/system_metrics.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+			Message:    fmt.Sprintf("system metrics for task id '%s' not found", opts.TaskID),
+		}
+	} else if err != nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    errors.Wrapf(err, "problem retrieving system metrics data for task id '%s'", opts.TaskID).Error(),
+		}
+	}
+
+	// check that the metric is valid so we can return the appropriate err
```
Comment (Kimchelly) [inline rest/data/system_metrics.go:34]: nit: error, not err

File: rest/data/system_metrics.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	// code.
+	_, ok := sm.Artifact.MetricChunks[metricType]
+	if !ok {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("metric type '%s' for task id '%s' not found", metricType, opts.TaskID),
+		}
+	}
+	sm.Setup(dbc.env)
```
Comment (Kimchelly) [inline rest/data/system_metrics.go:43]: Is there a reason we need to call `Setup` again when it was already called before this?

File: rest/data/system_metrics.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+///////////////////////////////
+// MockConnector Implementation
+///////////////////////////////
+
+func (mc *MockConnector) FindSystemMetricsByType(ctx context.Context, metricType string, opts dbModel.SystemMetricsFindOptions) (io.ReadCloser, error) {
+	var sm *dbModel.SystemMetrics
+	for key := range mc.CachedSystemMetrics {
+		val := mc.CachedSystemMetrics[key]
```
Comment (Kimchelly) [inline rest/data/system_metrics.go:62]: Can this just iterate `for key, val := range mc.CachedSystemMetrics`?

File: rest/data/system_metrics.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+		if opts.EmptyExecution {
+			if val.Info.TaskID == opts.TaskID && (sm == nil || val.Info.Execution > sm.Info.Execution) {
+				sm = &val
+			}
+		} else {
+			if val.Info.TaskID == opts.TaskID && val.Info.Execution == opts.Execution {
```
Comment (Kimchelly) [inline rest/data/system_metrics.go:68]: Can this just be combined with the `else` block to be `else if`?

File: rest/data/system_metrics.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+		}
+	}
+	if sm == nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("system metrics for task id '%s' not found", opts.TaskID),
+		}
+	}
+
+	// check that the metric is valid so we can return the appropriate err
```
Comment (Kimchelly) [inline rest/data/system_metrics.go:81]: nit: error, not err.

File: rest/data/system_metrics.go (hunk 3 above)
Comment (julianedwards) [reply rest/data/system_metrics.go:62]: If we use `key, val...` I can't use a pointer to `val` to assign it to `sm`. Iterating using `range` in go is weird.

**Pull request merged**